extensions/jfr/runtime/src/main/java/io/quarkus/jfr/runtime/http/rest/RestReactiveEndEvent.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.jfr.runtime.http.rest;
+
+import io.quarkus.jfr.runtime.SpanIdRelational;
+import io.quarkus.jfr.runtime.TraceIdRelational;
+import jdk.jfr.*;
+
+@Label("REST Reactive End")
+@Category({"Quarkus", "HTTP"})
+@Name("io.quarkus.rest.RestReactiveEnd")
+@Description("This event records information at the end of the REST API executed as reactive")
+@StackTrace(false)
+public class RestReactiveEndEvent extends Event {
```

Comment:
Yes. I agree with you.
If only a single process is being executed in a single thread, then yes indeed. This includes the case of virtual threads.

However, when non-blocking is implemented in virtual threads, multiple processes are being processed in a single thread apparently at the same time. Therefore, the normal way of recording events results in overlapping events in the JMC lanes. So, as a last resort, Start and End events are created to avoid overlap.

I wish I had an idea to separate JMC lanes for each request, like VirtualThread...

Reply:
Today there is a bug JMC where it uses the platform thread ID (which is always 0 for virtual threads) as index in the thread graph. Maybe that's the problem?

I think virtual threads should be filtered out in the thread graph. It's not possible to scroll a million lanes.

If they are to be included, perhaps after some filtering, the Java thread ID should be used as index, if it's a Java thread. For native threads, the index could be the negative value of the platform thread ID, so it never clashes with the Java thread ID.

Reply:
This will probably not help as the problem seems be related to VT, but it's possible to separate events into different lanes using the category annotation. See the example here: https://docs.oracle.com/en/java/javase/21/docs/api/jdk.jfr/jdk/jfr/Category.html

Reply:
@egahlin 
You are right, it would be impossible to scroll through a million lanes. However, it would also be impossible to see 1000 overlapping events. Different types of events can be viewed in different activity lanes, but not if they are the same event.

But it might be better if we first identify the prolonged request in the OTel or JMC event browser, focus on the ID that the request has, and then look at the threads, as I don't think the information from other requests would overlap.

First-time users will definitely be confused by the overlap. Therefore, I would need to describe this process in the documentation and make sure it is properly known.